### ISSUE TITLE
getCookieValue doc needs to clarify what its default filters are doing

### ISSUE THREAD

**Issue by jcranfordupgrade** [gatling/gatling#4534]
The getCookieValue API doc is misleading about what happens when optional parameters are omitted (i.e. withSecure, withPath).
- https://docs.gatling.io/reference/script/protocols/http/helpers/#getting-cookie-value

Example request:
```
GET https://example.com/apigateway/sessionstart?clientId=mobile-web
headers:
	user-agent: Gatling
	host: example.com
```
Example Response:
```
version:
	HTTP/1.1
status:
	302 Found
headers:
	Date: Fri, 01 Mar 2024 22:11:05 GMT
	Content-Length: 0
	location: https://example.com/mobile-web
	set-cookie: target-url=https://example.com/mobilestart; Path=/; Max-Age=3600; Expires=Sat, 02 Mar 2024 22:11:05 GMT; Secure; HttpOnly; SameSite=Strict
	set-cookie: JSESSIONID=ABC123ABC123; Max-Age=300; Expires=Fri, 1 Mar 2024 22:41:05 GMT; Path=/apigateway/sessionstart; Secure; HttpOnly; SameSite=Strict
```
Example code:
```
CoreDsl.scenario("MY_SCENARIO")
            .feed(ATTEMPTS.iterator())
            .feed(FEEDER_UUID)
            .exec(http("SESSION START")
                    .get("https://example.com/apigateway/sessionstart?clientId=mobile-web")
                    .disableFollowRedirect()
                    .check(status().is(302))
                    .check(header("set-cookie").findAll().saveAs("set-cookie"))
                    .check(bodyString().saveAs("BODY"))
            .exec(getCookieValue(CookieKey("JSESSIONID")))
            .exec(getCookieValue(CookieKey("target-url")))
            .exec(session -> {
                log.info(" ***** Start Session: "
                        + ", set-cookie: " + session.getString("set-cookie") // works, prints an array of 2 cookies
                        + ", JSESSIONID: " + session.getString("JSESSIONID") // doesn't work, prints null
                        + ", target-url: " + session.getString("target-url") // doesn't work, prints null
                        + ", body: " + session.getString("BODY"));
                return session;
            });
```

In the example code:
1. `header("set-cookie").findAll()` works
2. `getCookieValue(CookieKey("name"))` does not work
- need to add `.withSecure(true)` and `withPath("/apigateway/sessionstart")` to make it work for JSESSIONID
- need to add `.withSecure(true)` to make it work for target-url

***

For secure parameter, the doc says it is optional:
- `secure is optional, defaulting to false, means you only want secured cookies`

There are two different interpretations possible for the default use case:
1. No filtering => In other words, give me secure or insecure cookies.
2. Filter => In other words, only give me insecure cookies.

Intuitively, the default option is 1 in every other cookie API I have ever used. For whatever reason, the default in Gatling is 2, which is fine but only if it is documented correctly to prevent misinterpretation.

I wasted a lot of time trying to diagnose why your getCookieValue API would not return the values I saw in debug logs and returned by the headers API. Please fix the docs.

**Comment by Amerousful** [comment]
I've also faced this case when you don't understand what's going on, and why it doesn't extract, but I believe that just `getCookieValue(CookieKey("name"))` doesn't work for a special reason. 
I believe the issue is **how to differentiate** between cookies.
It's nothing out of the ordinary that a user has multiple cookies with the same name across different domains. Furthermore, it is possible for cookies to have the same `name` and domain, but vary in their `paths`.

**Comment by slandelle** [comment]
Indeed, that’s one very complicated thing people might not be aware of unless they really know how their application works or have read RFC6235.

Still, I think we could make things easier for the most common and easier case. PR incoming.

Cheers

**Comment by slandelle** [comment]
Closed as of #4536
